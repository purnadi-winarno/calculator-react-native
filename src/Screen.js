import React, { useCallback } from "react";
import { View, Text } from "react-native";
import Button from "./Button";
import styles from "./styles";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { NUMBERS, SIGNS } from "./constants";
import { onPressButton } from "./actions";

const Screen = ({ stack, onPressButton }) => {
  const onButtonPress = useCallback(text => onPressButton(text), []);
  function _renderNumbers() {
    return NUMBERS.map((numberRow, index) => (
      <View style={styles.numberRow} key={`row-${index}`}>
        {numberRow.map(number => (
          <Button
            text={number}
            square
            key={`button-${number}`}
            onPressHandler={onButtonPress}
          />
        ))}
      </View>
    ));
  }
  function _rendersSigns() {
    return SIGNS.map(sign => (
      <Button
        text={sign}
        key={`button-${sign}`}
        onPressHandler={onButtonPress}
      />
    ));
  }
  return (
    <View style={styles.container}>
      <View style={styles.inputBox}>
        <Text style={styles.inputText}>{stack.join("")}</Text>
      </View>
      <View style={styles.row}>
        <View style={styles.numberButtonContainer}>{_renderNumbers()}</View>
        <View style={styles.signButtonContainer}>{_rendersSigns()}</View>
      </View>
    </View>
  );
};

const mapStateToProps = state => ({
  stack: state.calculator.stack
});
const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      onPressButton
    },
    dispatch
  );
export default connect(mapStateToProps, mapDispatchToProps)(Screen);
