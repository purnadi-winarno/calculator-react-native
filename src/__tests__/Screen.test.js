import { shallow } from "enzyme";
import * as React from "react";
import Screen from "../Screen";
import { testStore } from "./libs";

const setUpComponent = (initialState = {}) => {
  const store = testStore(initialState);
  return shallow(<Screen store={store} />);
};

describe("Testing screen", () => {
  let wrapper;
  //let spy;

  beforeEach(() => {
    // spy = jest.fn();
    const initialState = {
      calculator: {
        stack: ["0"]
      }
    };
    wrapper = setUpComponent(initialState);
  });

  it("renders calculator screen correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
