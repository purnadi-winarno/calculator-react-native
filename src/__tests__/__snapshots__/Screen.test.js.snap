// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing screen renders calculator screen correctly 1`] = `
<ContextProvider
  value={null}
>
  <Screen
    onPressButton={[Function]}
    stack={
      Array [
        "0",
      ]
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
